pkg/controllers/disruption/orchestration/suite_test.go

```diff
@@ -0,0 +1,310 @@
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	. "knative.dev/pkg/logging/testing"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	disruptionevents "github.com/aws/karpenter-core/pkg/controllers/disruption/events"
+	"github.com/aws/karpenter-core/pkg/controllers/provisioning"
+	"github.com/aws/karpenter-core/pkg/utils/nodeclaim"
+
+	"github.com/aws/karpenter-core/pkg/apis"
+	"github.com/aws/karpenter-core/pkg/apis/settings"
+	"github.com/aws/karpenter-core/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-core/pkg/cloudprovider/fake"
+	"github.com/aws/karpenter-core/pkg/controllers/disruption/orchestration"
+	"github.com/aws/karpenter-core/pkg/controllers/state"
+	"github.com/aws/karpenter-core/pkg/controllers/state/informer"
+	"github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/operator/scheme"
+	"github.com/aws/karpenter-core/pkg/test"
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+
+	v1 "k8s.io/api/core/v1"
+	clock "k8s.io/utils/clock/testing"
+)
+
+var ctx context.Context
+var env *test.Environment
+var cluster *state.Cluster
+var cloudProvider *fake.CloudProvider
+var nodeStateController controller.Controller
+var nodeClaimStateController controller.Controller
+var fakeClock *clock.FakeClock
+var recorder *test.EventRecorder
+var queue *orchestration.Queue
+var prov *provisioning.Provisioner
+
+var replacements []nodeclaim.Key
+var ncKey nodeclaim.Key
+
+var nodeClaim1, nodeClaim2, replacementNodeClaim *v1beta1.NodeClaim
+var nodePool *v1beta1.NodePool
+var node1, node2, replacementNode *v1.Node
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Disruption/Orchestration")
+}
+
+var _ = BeforeSuite(func() {
+	env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...))
+	ctx = settings.ToContext(ctx, test.Settings(settings.Settings{DriftEnabled: true}))
+	fakeClock = clock.NewFakeClock(time.Now())
+	cloudProvider = fake.NewCloudProvider()
+	cluster = state.NewCluster(fakeClock, env.Client, cloudProvider)
+	nodeStateController = informer.NewNodeController(env.Client, cluster)
+	nodeClaimStateController = informer.NewNodeClaimController(env.Client, cluster)
+	recorder = test.NewEventRecorder()
+	prov = provisioning.NewProvisioner(env.Client, env.KubernetesInterface.CoreV1(), recorder, cloudProvider, cluster)
+	queue = orchestration.NewTestingQueue(env.Client, recorder, cluster, fakeClock, prov)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = BeforeEach(func() {
+	recorder.Reset() // Reset the events that we captured during the run
+
+	fakeClock.SetTime(time.Now())
+	cluster.Reset()
+	cloudProvider.Reset()
+	cloudProvider.InstanceTypes = fake.InstanceTypesAssorted()
+	cluster.MarkUnconsolidated()
+	queue.Reset()
```

Comment:
Why not just make a new queue?

Reply:
It essentially does that. If we make a new queue, the constructor for the disruption controller needs to be updated with the new queue.